src/model.ts

```diff
@@ -0,0 +1,54 @@
+export class Directive {
+  name: string;
+  arguments: DirectiveArgument[] = [];
+}
+
+export class DirectiveArgument {
+  name: string;
+  value: string;
+}
+
+export class Field {
+  directives?: Directive[] = [];
+  name: string;
+  type?: Type;
+  isRequired?: boolean;
+}
+
+export class Scalar {
+  name?: string;
+  description?: string;
+  value: string;
+}
+
+export class Class {
+  directives?: Directive[];
```

Comment:
I would probably get rid of all optional fields.  Or look into using `interfaces` instead.